src/test/java/ArraySortedBagTest.java

```diff
@@ -1,355 +1,381 @@
-        assertEquals(0, bag.size());
-    }
-
-    @Test
-    void nonEmptySortedBagIsEmptyReturnsFalse() {
-        SortedBag<Integer> bag = new ArraySortedBag<>();
-        bag.add(99);
-        assertFalse(bag.isEmpty());
-    }
-
-    @Test
-    void containsAddedElementToSortedBagReturnsTrue() {
-        SortedBag<Integer> bag = new ArraySortedBag<>();
-        bag.add(99);
-        assertTrue(bag.contains(99));
-    }
-
-    @Test
-    void containsMultipleExistingElementToSortedBagReturnsTrue() {
-        SortedBag<Integer> bag = new ArraySortedBag<>();
-        bag.add(99);
-        bag.add(99);
-        bag.add(99);
-        assertTrue(bag.contains(99));
-    }
-
-    @Test
-    void containsNonAddedElementToSortedBagReturnsFalse() {
-        SortedBag<Integer> bag = new ArraySortedBag<>();
-        bag.add(99);
-        assertFalse(bag.contains(101));
-    }
-
-    @Test
-    void getCountOfElementWithSingleCopyInSortedBagReturnsCorrectCount() {
-        SortedBag<Integer> bag = new ArraySortedBag<>();
-        bag.add(0);
-        bag.add(99);
-        bag.add(1);
-        bag.add(99);
-        bag.add(2);
-        bag.add(99);
-        assertEquals(1, bag.count(1));
-    }
-
-    @Test
-    void getCountOfElementWithMultipleCopiesInSortedBagReturnsCorrectCount() {
-        SortedBag<Integer> bag = new ArraySortedBag<>();
-        bag.add(0);
-        bag.add(99);
-        bag.add(1);
-        bag.add(99);
-        bag.add(2);
-        bag.add(99);
-        assertEquals(3, bag.count(99));
+    @BeforeEach
+    void createStack() {
+        classUnderTest = new ArraySortedBag<>();
     }
 
-    @Test
-    void getCountOfElementNotInSortedBagReturnsZero() {
-        SortedBag<Integer> bag = new ArraySortedBag<>();
-        bag.add(0);
-        bag.add(99);
-        bag.add(1);
-        bag.add(99);
-        bag.add(2);
-        bag.add(99);
-        assertEquals(0, bag.count(101));
-    }
+    @Nested
+    class WhenNewEmpty {
 
-    @Test
-    void addingElementsToSortedBagAreAddedInCorrectOrder() {
-        SortedBag<Integer> bag = new ArraySortedBag<>();
-        bag.add(5);
-        bag.add(1);
-        bag.add(2);
-        bag.add(0);
-        bag.add(4);
-        bag.add(3);
-        for (int i = 0; i < 6; ++i) {
-            assertEquals(i, bag.removeFirst());
+        @Test
+        void add_successfulAdd_returnsTrue() {
+            assertTrue(classUnderTest.add(0));
         }
-    }
 
-    @Test
-    void addingBeyondCapacityOfSortedBagCallsExpandCapacityToMakeRoomWhileMaintainingOrdering() {
-        SortedBag<Integer> bag = new ArraySortedBag<>(5);
-        bag.add(5);
-        bag.add(1);
-        bag.add(2);
-        bag.add(0);
-        bag.add(4);
-        bag.add(3);
-        for (int i = 0; i < 6; ++i) {
-            assertEquals(i, bag.removeFirst());
+        @Test
+        void remove_empty_throwsNoSuchElementException() {
+            assertThrows(NoSuchElementException.class, () -> classUnderTest.remove(0));
```

Comment:
This one is a little funny since there are two ways a `NoSuchElementException` could be thrown. They have different messages. 

I _could_ check the messages to differentiate since the messages will differ if it's an exception due to being empty vs. exception due to non existing. But also, that's work.

Reply:
Doesn't really matter imo. The important bit to test is the exception is thrown, not the exception from a specific line is actually thrown. Test the interface, not the implementation.